qemu/cpus.c

```diff
@@ -191,15 +169,10 @@ static int qemu_tcg_init_vcpu(CPUState *cpu)
         uc->tcg_halt_cond = cpu->halt_cond;
         snprintf(thread_name, VCPU_THREAD_NAME_SIZE, "CPU %d/TCG",
                 cpu->cpu_index);
-        if (qemu_thread_create(uc, cpu->thread, thread_name, qemu_tcg_cpu_thread_fn,
-                cpu, QEMU_THREAD_JOINABLE))
-            return -1;
+        qemu_thread_get_self(uc, cpu->thread);
```

Comment:
I want to be able to emulate multi-threaded code, but right now my best plan is to thread my host program and schedule `uc_emu_start()` threads manually.

I think Unicorn would do a much better job than me with cache coherency, as I have no idea what a memory fence/barrier/etc would do across two Unicorn instances with `mem_map_ptr`-shared memory.